storage: hide kv.raft_log.synchronize setting

tbg opened the pull request:
Also reword the help text to mention the potential for data loss.

Fixes #26282.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/26418)
<!-- Reviewable:end -->

bdarnell commented:
I generally dislike hidden settings - it hides them from `show all cluster settings` and our docs, but what about third-party guides or cargo-culted configurations? I'd rather rename this to have `unsafe` in its name or something like that.

tschottdorf commented:
@bdarnell we can also agree to do nothing here, but I'd rather not rename the setting, that's just asking for trouble.

tbg committed:
storage: clarify kv.raft_log.synchronize setting

Reword the help text to mention the potential for data loss.

Fixes #26282.

Release note: None

tschottdorf commented:
SGTM, done.

bors r=bdarnell
